### PR TITLE
Remove mangling

### PR DESCRIPTION
**Pull request by ljacobson64**
This PR removes the need for name mangling in `meshtal_funcs.h` by making use of `bind(c)` in `fmesh_mod.F90`. Previously, there was a macro in `meshtal_funcs.h` that attempted to determine the correct name mangling, but I've found situations where it didn't work properly. This is a much simpler way to do it.

(spoilers)

The real reason I did this is that MCNP6.2 uses Fortran2003, and from what I can tell, Intel's name mangling scheme for Fortran2003 is slightly different than it is for Fortran90. (It could be caused by something else, but regardless, the mangling is different between MCNP6.2 and all other versions of MCNP.) Trying to fix the name mangling macro would be pretty ugly because a different name mangling would need to be used for DAG-MCNP5 (Fortran90) than DAG-MCNP6.2 (Fortran2003) when trying to build both at the same time. This PR completely removes that problem, and it will remove any future problems related to name mangling as well.

I've gotten DAG-MCNP6.2 working (or at least compiling) on top of these changes, but support for that will come in a future PR.

### CODE REVIEW

**Comment by pshriwise** [comment]
This is great! I agree - this is the right way to handle name mangling long-term. Will review shortly (once Travis has completed).

File: src/mcnp/meshtal_funcs.h
Comment (pshriwise) [inline src/mcnp/meshtal_funcs.h:42]: Good to see this go...

File: news/PR-0556.rst
Comment (pshriwise) [inline news/PR-0556.rst:5]: Can you add renaming of the patch files here? This will be important for external users.

**Review by pshriwise — COMMENTED** [review]
LGTM! Just one small request. 

Will wait until the end of the day on Monday to merge so others can take a look.

File: news/PR-0556.rst
Comment (ljacobson64) [reply news/PR-0556.rst:5]: Good call. Done.

**Review by pshriwise — APPROVED** [review]
Thanks for the updates @ljacobson64! LGTM

**Pull request merged**
